airflow/providers/amazon/aws/operators/athena.py

```diff
@@ -73,19 +82,17 @@ def __init__(  # pylint: disable=too-many-arguments
         self.result_configuration = result_configuration or {}
         self.sleep_time = sleep_time
         self.max_tries = max_tries
-        self.query_execution_id = None
-        self.hook = None
+        self.query_execution_id = None  # type: Optional[str]
+        self.hook = self.get_hook()
```

Comment:
This should be None bye default to avoid initializing hooks in Operators init.

Otherwise, whenever the scheduler parses the DAG files, DB calls are made to the database.

You could do the following to fix mypy errors:

```suggestion
        self.hook: Optional[AWSAthenaHook] = None
```

Reply:
Using `self.hook: Optional[AWSAthenaHook] = None` results in:
```
airflow/providers/amazon/aws/operators/athena.py:127: error: Item "None" of
"Optional[AWSAthenaHook]" has no attribute "stop_query"
                response = self.hook.stop_query(self.query_execution_id)
```